model_compression_toolkit/keras/quantizer/mixed_precision/input_layer_quantize_transform.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2022 Sony Semiconductors Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import keras.layers
+from keras.engine.input_layer import InputLayer
+from tensorflow_model_optimization.python.core.quantization.keras.graph_transformations import transforms
+
+from tensorflow_model_optimization.python.core.quantization.keras.quantize_wrapper import QuantizeWrapper
+
+from model_compression_toolkit.keras.quantizer.mixed_precision.quantization_config_factory import \
+  quantization_config_builder_mixed_precision
+from model_compression_toolkit.keras.quantizer.mixed_precision.selective_quantize_config import SelectiveQuantizeConfig
+
+
+class InputLayerQuantizeTransform(transforms.Transform):
+    """
+    Quantizes InputLayer, by adding QuantizeLayer after it.
+    InputLayer => InputLayer -> QuantizeLayer
+    """
```

Comment:
Maybe we should inherit from Tensorflow input transform, instead of general transform, so we can save the pattern method implementation, and so it will be more obvious that we based our implementation on their transformation?
@haihabi what do you think?

Reply:
@tocker Hi Amir, your advice is needed. We debate over the similarity of our [file](https://github.com/sony/model_optimization/pull/143/files/8d51cd66e043f5ccab6e95ff5475ffa75e0a9fc2#diff-67092abf8b5252d6daad897277a71a1feeb051d4cb657a35cf70a5c94e015d9d) comparing to Tensorflow [file:](https://github.com/tensorflow/model-optimization/blob/master/tensorflow_model_optimization/python/core/quantization/keras/default_8bit/default_8bit_transforms.py#L579)